### DISCUSSION TITLE
Signed Exchanges (SXGs) Support

### DISCUSSION THREAD

**Discussion by slavakurilyak** [vercel/next.js#37640] [Ideas]
It would be great to see Next.js or Vercel integrate with signed exchanges.

Introduced by Google in 2020, [Signed Exchanges (SXGs)](https://web.dev/signed-exchanges/) allow developers to accomplish the following:

- Improve your website's performance by making cacheable resources available on Google’s Signed Exchanges
- Enable Chromium-based browsers to prefetch your website on Google's search results page and make your website faster
- Improve the Largest Contentful Paint (LCP) which is part of the Core Web Vitals and increase your SEO ranking

To get started, check out the [platform-specific SXG tooling](https://web.dev/signed-exchanges/#platform-specific-sxg-tooling), [general-purpose SXG tooling](https://web.dev/signed-exchanges/#general-purpose-sxg-tooling) and [SXG libraries](https://web.dev/signed-exchanges/#sxg-libraries).

**Comment by slavakurilyak** [comment]
Google provides a tool called [amppackager](https://github.com/ampproject/amppackager) to improve AMP URLs via Signed Exchanges. By running [AMP](https://nextjs.org/docs/api-reference/next/amp) pages in a proper configuration, web publishers enable origin URLs to appear in AMP search results.

**Comment by YoavRheims** [comment]
A must-have and now starting to be supported on many popular browsers: https://caniuse.com/sxg

**Comment by sigginjals** [comment]
Any updates here? @leerob, Is this something you are looking at?

**Comment by pepawel** [comment]
In my experience, one of the key challenges with Signed Exchanges is ensuring subresource prefetching works correctly. This is especially true for Next.js. I’ve written about this in detail in the second part of my series:

https://blog.pawelpokrywka.com/p/subresources-prefetching-with-signed-exchanges

Hope this adds value to the conversation!
